core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java

```diff
@@ -188,6 +189,15 @@ public void validate() {
       boolean preload = attributes.attribute(PRELOAD).get();
       boolean transactional = attributes.attribute(TRANSACTIONAL).get();
       ConfigurationBuilder builder = getBuilder();
+
+      Class storeClazz = attributes.getKlass();
+      if (storeClazz != null && storeClazz.isAnnotationPresent(Store.class)) {
```

Comment:
@wburns I agree, however I think that we can make this more generic as I envisage that long term we can use the `@Store` attribute to validate other store properties. So I have made it so that if the `@Store` annotation is missing a generic warning is issued stating that it is not possible for all validation to occur when said annotation is missing.